day23/stringsRearrangement.py

```diff
@@ -0,0 +1,37 @@
+from itertools import permutations
+
+def stringsRearrangement(inputArray):
+    for perm in permutations(inputArray):
+        if neighborsDifferByOne(perm):
+            return True 
+    return False
+
+def neighborsDifferByOne(inputArray):
+    if len(inputArray) <= 1:
+        return True
+    return differByOne(inputArray[0], inputArray[1]) and neighborsDifferByOne(inputArray[2:])
+
+def differByOne(word, anotherWord):
+    if len(word) != len(anotherWord):
+        return False
+
+    differencesCount = 0
+    for i in xrange(len(word)):
+        if word[i] != anotherWord[j]:
+            differencesCount += 1
+
+    if differencesCount == 0 or differencesCount >= 2:
+        return False
+
+    return True
+
+
+def main():
+    assert stringsRearrangement([], True)
```

Comment:
you should start testing your individual units/functions instead of doing one test for the overall program